File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.util.*;
import java.io.*;

public class Main {
	
	public static void main(String[] args) throws IOException{
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String str = br.readLine();
		
		int result = 0;
		
		int num = 0;
		Boolean minusOn = false;
		for(int i=0;i<str.length();i++) {
			char ch = str.charAt(i);
			if(Character.isDigit(ch)) {
				num = num*10+Integer.parseInt(""+ch);
			}
			else if(ch == '-') {
				if(minusOn) {
					result -= num;
				}
				else {
					result += num;
				}
					
				
				minusOn = true;
				num =0;
			}
			else if(ch == '+')
			{
				if(minusOn) {
					result -= num;
					
				}
				else
					result+= num;
				num =0;
			}
		}
		if(minusOn)
			result -= num;
		else
			result+=num;
		
		System.out.println(result);
	}

}

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
import sys
input = sys.stdin.readline

equation = input().rstrip()

# 결과가 작으려면, -가 나오면 그 뒤를 다 마이너스로 만들수있다?
num = ""
minus_flag = False
result = 0
for i in equation:
    if '0'<=i<='9' : 
        num+= i
    else:
        if minus_flag:
            result -= int(num)
        else:
            result += int(num)
        if i == '-':
            minus_flag = True
        num = ""
if minus_flag: result-= int(num)
else: result+= int(num)

print(result)